wallet/wallet.c

```diff
@@ -3408,6 +3408,13 @@ static bool wallet_stmt2htlc_in(struct channel *channel,
 	/* FIXME: save path_key in db !*/
 	in->path_key = NULL;
 	in->payload = NULL;
+	/* FIXME: save extra_tlvs in db! But: check the implications that a
+	 * spammy peer - giving us big extra tlvs - would have on our database.
+	 * Right now, not saving the extra tlvs in the db seems OK as it is
+	 * only relevant in the case that I forward but restart in the middle
+	 * of a payment.
+	 */
```

Comment:
Notice that messages in the LN protocol are limited to 65KiB, and `update_add_htlc` contains at least a 1365B onion, so 64KiB is the natural limit for the extra_tlvs (and there are other fields that constrain further), as such a limit on pending HTLCs is likely sufficient to also limit the DB space used for extra TLVs.